File: replace-all-s-to-avoid-consecutive-repeating-characters/replace-all-s-to-avoid-consecutive-repeating-characters.cpp
class Solution {
public:
    string modifyString(string s) {
        if(s.size() == 1 && s[0] == '?')return "a";
        for(int i = 0 ; i < s.size() ; i++){
            if(s[i] == '?'){
                if(i == 0){
                    for(int p = 0 ; p < 26 ; p++){
                        if(s[i+1] != 'a' + p){
                            s[i] = 'a'+p;
                        }
                    }
                }else if(i == s.size() - 1){
                    for(int p = 0 ; p < 26 ; p++){
                        if(s[i-1] != 'a' + p){
                            s[i] = 'a'+p;
                        }
                    }
                }else{
                    for(int p = 0 ; p < 26 ; p++){
                        if(s[i-1] != 'a' + p && s[i+1] != 'a' + p ){
                            s[i] = 'a'+p;
                        }
                    }
                }
            }
        }
        return s;
    }
};

File: replace-all-s-to-avoid-consecutive-repeating-characters/replace-all-s-to-avoid-consecutive-repeating-characters.go
func modifyString(p string) string {
    if len(p) == 1 && rune(p[0]) == '?' {
        return "a"
    }
    s := []rune(p)
    length := len(s)
    for i := 0 ; i < length ; i++ {
        if s[i] == '?' {
            if i == 0 {
                for j := 0 ; j < 26 ; j++{
                    if s[i+1] != rune('a'+j) {
                        s[i] = rune('a'+j)
                    }
                }
            }else if i == length - 1 {
                for j := 0 ; j < 26 ; j++{
                    if s[i-1] != rune('a'+j) {
                        s[i] =rune('a'+j)
                    }
                }
            }else{
                for j := 0 ; j < 26 ; j++{
                    if s[i-1] != rune('a'+j) && s[i+1] != rune('a'+j) {
                        s[i] = rune('a'+j)
                    }
                }
            }
        }
    }
    
    return string(s)
}